File: Digito Verificador/Modulo 11 Padrao/Teste - Arquivo 11/GerarListaMatriculasMod11.dart
import 'dart:io';

void removerCaractere(List<String> CADEIA, String CARACTERE) {
  int POS1 = 0;
  int POS2 = 0;
  while (CADEIA[POS1] != '\0') {
    CADEIA[POS2] = CADEIA[POS1];
    if (CADEIA[POS2] != CARACTERE) {
      POS2 += 1;
    }
    POS1 += 1;
  }
  CADEIA[POS2] = '\0';
}

String mod11(List<String> MATRIC_CPT) {
  List<String> MATRIC_SPT = List<String>.filled(8, '\0');
  List<int> MATRIC_INT = List<int>.filled(6, 0);
  int SOMA = 0;
  String DV = ' ';
  for (int I = 0; I < 8; I++) {
    MATRIC_SPT[I] = MATRIC_CPT[I];
  }
  removerCaractere(MATRIC_SPT, '.');
  for (int I = 0; I < 6; I++) {
    MATRIC_INT[I] = MATRIC_SPT[I].codeUnitAt(0) - 48;
  }
  for (int I = 0; I < 6; I++) {
    SOMA += MATRIC_INT[I] * (7 - I);
  }
  if (SOMA % 11 == 0) {
    DV = '0';
  } else if (SOMA % 11 == 1) {
    DV = 'X';
  } else if (SOMA % 11 == 10) {
    DV = '1';
  } else {
    DV = String.fromCharCode(11 - (SOMA % 11) + 48);
  }
  return DV;
}

void main() async {
  List<String> MATRICULA = List<String>.filled(12, '\0');

  File arquivo = File('numeros.txt');
  IOSink fileSink;
  try {
    fileSink = arquivo.openWrite();
  } catch (e) {
    stderr.writeln('Erro ao abrir o arquivo.');
    stdout.writeln('Programa encerrado.');
    stdout.writeln('');
    return;
  }

  stdout.writeln('GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM "DV"');
  stdout.writeln('Padrao: Modulo 11');
  stdout.writeln('---------------------------------------------------------------');
  stdout.writeln('');

  stdout.write('Aguarde arquivo sendo gerado... ');

  for (int I = 0; I <= 999999; I++) {
    String MATRICULA_STR = I.toString().padLeft(6, '0');
    for (int J = 0; J < MATRICULA_STR.length; J++) {
      MATRICULA[J] = MATRICULA_STR[J];
    }
    String DV = mod11(MATRICULA);
    String MATRICULA_FORMATADA =
        '${MATRICULA_STR.substring(0, 3)}.${MATRICULA_STR.substring(3)}-$DV';
    fileSink.writeln(MATRICULA_FORMATADA);
  }

  await fileSink.close();

  stdout.writeln('\n\nSUCESSO: Arquivo gerado.\n');
}

File: Digito Verificador/Modulo 11 Padrao/Teste - Arquivo 11/GerarListaMatriculasMod11.py
import sys

def removerCaractere(CADEIA, CARACTERE):
    POS1 = 0
    POS2 = 0
    while CADEIA[POS1] != '\0':
        CADEIA[POS2] = CADEIA[POS1]
        if CADEIA[POS2] != CARACTERE:
            POS2 += 1
        POS1 += 1
    CADEIA[POS2] = '\0'

def mod11(MATRIC_CPT):
    MATRIC_SPT = ['\0'] * 8
    MATRIC_INT = [0] * 6
    SOMA = 0
    DV = ' '
    for i in range(8):
        MATRIC_SPT[i] = MATRIC_CPT[i]
    removerCaractere(MATRIC_SPT, '.')
    for i in range(6):
        MATRIC_INT[i] = ord(MATRIC_SPT[i]) - 48
    for i in range(6):
        SOMA += MATRIC_INT[i] * (7 - i)
    if SOMA % 11 == 0:
        DV = '0'
    elif SOMA % 11 == 1:
        DV = 'X'
    elif SOMA % 11 == 10:
        DV = '1'
    else:
        DV = chr(11 - (SOMA % 11) + 48)
    return DV

def main():
    MATRICULA = ['\0'] * 12

    try:
        ARQUIVO = open("numeros.txt", "w")
    except IOError:
        sys.stderr.write("Erro ao abrir o arquivo.\n")
        print("Programa encerrado.")
        print()
        return

    print("GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"")
    print("Padrao: Modulo 11")
    print("---------------------------------------------------------------")
    print()

    print("Aguarde arquivo sendo gerado... ", end="")
    sys.stdout.flush()

    for I in range(1000000):
        MATRICULA_STR = "{:06d}".format(I)
        for i in range(len(MATRICULA_STR)):
            MATRICULA[i] = MATRICULA_STR[i]
        DV = mod11(MATRICULA)
        MATRICULA_FORMATADA = "{}.{}-{}".format(MATRICULA_STR[:3], MATRICULA_STR[3:], DV)
        ARQUIVO.write(MATRICULA_FORMATADA + "\n")

    ARQUIVO.close()

    print("\n\nSUCESSO: Arquivo gerado.\n")

if __name__ == "__main__":
    main()
